lib/vSphere/driver.rb

```diff
@@ -0,0 +1,920 @@
+					else
+						ipAddress = vm.guest.net[config.management_network_adapter_slot].ipConfig.ipAddress.detect { |addr| addr.origin != 'linklayer' }
+					end
+
+					return nil if ipAddress.nil?
+ 					return ipAddress.ipAddress
+				else
+					return config.network_adapters[config.management_network_adapter_slot].ip_address.to_s if config.network_adapters[config.management_network_adapter_slot].ip_address.is_a?(IPAddr)
+					return config.network_adapters[config.management_network_adapter_slot].ip_address
+				end
+			end
+
+			def get_datacenter(connection, machine)
+				connection.serviceInstance.find_datacenter(machine.provider_config.data_center_name) || fail(Errors::VSphereError, :missing_datacenter)
+			end
+
+			def get_vm_by_uuid(connection, machine)
+				get_datacenter(connection, machine).vmFolder.findByUuid machine.id
+			end
+
+			def get_resource_pool(datacenter, machine)
+				rp = get_compute_resource(datacenter, machine)
+
+				resource_pool_name = machine.provider_config.resource_pool_name || ''
+
+				entity_array = resource_pool_name.split('/')
+				entity_array.each do |entity_array_item|
+					next if entity_array_item.empty?
+					if rp.is_a? RbVmomi::VIM::Folder
+						rp = rp.childEntity.find { |f| f.name == entity_array_item } || fail(Errors::VSphereError, :missing_resource_pool)
+					elsif rp.is_a? RbVmomi::VIM::ClusterComputeResource
+						rp = rp.resourcePool.resourcePool.find { |f| f.name == entity_array_item } || fail(Errors::VSphereError, :missing_resource_pool)
+					elsif rp.is_a? RbVmomi::VIM::ResourcePool
+						rp = rp.resourcePool.find { |f| f.name == entity_array_item } || fail(Errors::VSphereError, :missing_resource_pool)
+					elsif rp.is_a? RbVmomi::VIM::ComputeResource
+						rp = rp.resourcePool.find(resource_pool_name) || fail(Errors::VSphereError, :missing_resource_pool)
+					else
+						fail Errors::VSphereError, :missing_resource_pool
+					end
+				end
+				rp = rp.resourcePool if !rp.is_a?(RbVmomi::VIM::ResourcePool) && rp.respond_to?(:resourcePool)
+				rp
+			end
+
+			def get_compute_resource(datacenter, machine)
+				cr = find_clustercompute_or_compute_resource(datacenter, machine.provider_config.compute_resource_name)
+				fail Errors::VSphereError, :missing_compute_resource if cr.nil?
+				cr
+			end
+
+			def find_clustercompute_or_compute_resource(datacenter, path)
+				if path.is_a? String
+					es = path.split('/').reject(&:empty?)
+				elsif path.is_a? Enumerable
+					es = path
+				else
+					fail "unexpected path class #{path.class}"
+				end
+				return datacenter.hostFolder if es.empty?
+				final = es.pop
+
+				p = es.inject(datacenter.hostFolder) do |f, e|
+					f.find(e, RbVmomi::VIM::Folder) || return
+				end
+
+				begin
+					if (x = p.find(final, RbVmomi::VIM::ComputeResource))
+						x
+					elsif (x = p.find(final, RbVmomi::VIM::ClusterComputeResource))
+						x
+					end
+				rescue Exception
```

Comment:
Ah, I see. It's too bad that it's difficult to tell in git when code is just moved around like that. 

Looks like it passed rubocop before because `vim_helpers.rb` was specifically excluded in the rubocop configuration: https://github.com/nsidc/vagrant-vsphere/blob/master/.rubocop.yml#L7

So, no need to worry about this then! I'll just update the config to point to name the correct file.